### PR TITLE
Adding support to sftp protocol when you want things more private and secure.

### PR DESCRIPTION
**Pull request by pankajagarwal**

### CODE REVIEW

**Comment by nanliu** [comment]
Please add a spec tests and I'll merge it.

**Comment by pankajagarwal** [comment]
spec test added and is passing.

File: manifests/file.pp [hunk 1]
```diff
@@ -131,6 +131,13 @@
         command   => $command,
       }
     }
+    /^sftp:\/\//: {
+      $actualpath = regsubst("${source}",'^sftp:\/\/', '')
+      $command = "sftp -o StrictHostKeyChecking=no ${actualpath} ${target_file}"
```
Comment (nanliu) [inline manifests/file.pp]: stricthostkeychecking should be a configurable option.

**Comment by jyaworski** [comment]
Hello:

What's the status of this PR? Can it be squashed?

**Comment by bastelfreak** [comment]
I will close this PR due to lack of response. Feel free to reopen/rebase if you still want it merged.

**Pull request closed without merging**